x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/TranslatorHandler.java

```diff
@@ -21,7 +25,12 @@
 
     Query asQuery(Expression e);
 
-    Query wrapFunctionQuery(ScalarFunction sf, Expression field, Query q);
+    default Query wrapFunctionQuery(ScalarFunction sf, Expression field, Supplier<Query> querySupplier) {
+        if (field instanceof FieldAttribute) {
+            return ExpressionTranslator.wrapIfNested(querySupplier.get(), field);
+        }
+        return new ScriptQuery(sf.source(), sf.asScript());
+    }
 
     String nameOf(Expression e);
```

Comment:
Loosely coupled to this refactoring, but: couldn't `nameOf()` also be offered a static implementation here? All implementations could actually have it as a static method and SQL's (whose implementation lives actually in `QueryTranslator`) only treats the `DateTimeFunction` case additionally, compared to QL's.
So a bit more duplication could be stripped away.